airbyte-cdk/bulk/toolkits/load-object-storage/src/test/kotlin/io/airbyte/cdk/load/file/object_storage/ObjectStoragePathFactoryUTest.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.file.object_storage
+
+import io.airbyte.cdk.load.command.DestinationStream
+import io.airbyte.cdk.load.command.object_storage.ObjectStoragePathConfiguration
+import io.airbyte.cdk.load.command.object_storage.ObjectStoragePathConfigurationProvider
+import io.airbyte.cdk.load.file.TimeProvider
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class ObjectStoragePathFactoryUTest {
+    @MockK lateinit var stream: DestinationStream
+    @MockK lateinit var pathConfigProvider: ObjectStoragePathConfigurationProvider
+    @MockK lateinit var timeProvider: TimeProvider
+
+    @BeforeEach
+    fun setup() {
+        every { stream.descriptor } returns DestinationStream.Descriptor("test", "stream")
+        every { timeProvider.syncTimeMillis() } returns 0
+        every { timeProvider.currentTimeMillis() } returns 1
+    }
+
+    @Test
+    fun `test matcher with suffix`() {
+        every { pathConfigProvider.objectStoragePathConfiguration } returns
+            ObjectStoragePathConfiguration(
+                "prefix",
+                null,
+                "path/",
+                "ambiguous_filename",
+                false,
+            )
+        val factory = ObjectStoragePathFactory(pathConfigProvider, null, null, timeProvider)
+
+        val matcher = factory.getPathMatcher(stream, "(-\\d+)?")
+        val match1 = matcher.match("prefix/path/ambiguous_filename")
+        assert(match1 != null)
```

Comment:
Nit, should we use assertNonNull/assertNull?

Reply:
sure